### PR TITLE
Feature/integration tests

### PR DESCRIPTION
**Pull request by vgarvardt**
* Integration cucumber-like tests
* Fixed some API endpoints while working on tests - `Location` header for registering new proxies, error handling and error codes for several API calls
* Added custom `prefix` support for `redis` storage and some docs on this
* Fixed `chi` failing with panic when proxies list is empty

### CODE REVIEW

File: pkg/api/error.go [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+	"github.com/hellofresh/janus/pkg/errors"
+)
 
 var (
 	// ErrAPIDefinitionNotFound is used when the api was not found in the datastore
 	ErrAPIDefinitionNotFound = errors.New(http.StatusNotFound, "api definition not found")
 
 	// ErrAPINameExists is used when the API name is already registered on the datastore
-	ErrAPINameExists = errors.New(http.StatusBadRequest, "api name is already registered")
+	ErrAPINameExists = errors.New(http.StatusConflict, "api name is already registered")
```
Comment (rafaeljesus) [inline pkg/api/error.go:14]: 👍

File: docs/quick_start/storage.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Storage
+
+Some Janus functionality requires key/value storage. This functionality is:
```
Comment (italolelis) [inline docs/quick_start/storage.md]: `This functionalities are`

**Review by italolelis — CHANGES_REQUESTED** [review]
Can we run these tests on travis?

File: pkg/api/repository.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package api
+
+// Repository defines the behaviour of a proxy specs repository
+type Repository interface {
```
Comment (rafaeljesus) [inline pkg/api/repository.go:4]: 👍

**Review by rafaeljesus — APPROVED** [review]

**Comment by rafaeljesus** [comment]
That issue regarding `COVERALLS_TOKEN` again on travis

**Comment by italolelis** [comment]
@rafaeljesus it's not that issue, it's something else

**Comment by vgarvardt** [comment]
Yes, I can reproduce locally - problem with `overalls` failing to process `godog_test.go` in project root. Trying to fix this.

**Comment by vgarvardt** [comment]
Fixed `overalls` issue and one more I found with `chi`.

@italolelis running theses tests on travis is possible, but requires some more work. There are two things that should be solved:
* run `docker-compose` in travis environment, that does not seem to be big problem - https://elliot.land/post/using-docker-compose-on-travis-ci
* run travis build after new release is built or build artifact after successful unit tests and run integration tests against this artifact

**Review by italolelis — APPROVED** [review]

**Comment by italolelis** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/147/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway) [Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/147/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway) [Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/147/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Pull request merged**